### PR TITLE
types,statistics: don't use time.Local timezone in time Sub

### PR DESCRIPTION
**Pull request by tiancaiamao**
```
package main

import (
	"fmt"
	"time"
)

func main() {
	t1 := time.Date(2017, 1, 1, 0, 0, 0, 0, time.Local)
	t2 := time.Date(2017, 4, 1, 0, 0, 0, 0, time.Local)
	fmt.Println(t1, t2)
}
```

If the system time zone is something other than Beijing, the result is very strange (Denver for example):
```
2017-01-01 00:00:00 -0700 MST
2017-04-01 00:00:00 -0600 MDT
```

The time zone changed!  This cause the time.Sub() method return a wrong result and one of the statistics test failed.

@lamxTyler

### CODE REVIEW

File: types/time.go [hunk 1]
```diff
@@ -502,10 +502,9 @@ func (t *Time) Check() error {
 func (t *Time) Sub(t1 *Time) Duration {
 	var duration gotime.Duration
 	if t.Type == mysql.TypeTimestamp && t1.Type == mysql.TypeTimestamp {
-		// TODO: Consider time_zone variable.
-		a, err := t.Time.GoTime(gotime.Local)
+		a, err := t.Time.GoTime(t.TimeZone)
```
Comment (lamxTyler) [inline types/time.go]: Can we make sure that the `t.TimeZone` is always not nil?

Comment (tiancaiamao) [reply types/time.go]: It should not be nil for timestamp... I think I can...
/run-all-tests
to check it.

Comment (shenli) [reply types/time.go]: Could you add a test case for this?

Comment (tiancaiamao) [reply types/time.go]: Not in this PR, because it needs to change the system timezone to reproduce it. @shenli

**Comment by shenli** [comment]
https://golang.org/src/time/time.go?s=39433:39516#L1295

**Comment by shenli** [comment]
LGTM

**Pull request merged**